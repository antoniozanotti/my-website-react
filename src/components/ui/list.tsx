import { ReactNode } from "react";

export interface ListProps {
  className?: string;
  children?: ReactNode;
}

export default function List({
  children,
  className = ""
}: ListProps) {
  return (
    <ul className={`list-disc list-inside ${className}`}>
      {children}
    </ul>
  );
}
