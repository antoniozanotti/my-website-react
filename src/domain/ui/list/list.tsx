export function List({
  children,
  ...props
}: React.ComponentProps<"ul">) {
  const baseStyles = "list-disc list-inside";
  props.className = props.className
    ? `${baseStyles} ${props.className}`
    : baseStyles;

  return <ul {...props}>{children}</ul>;
}
