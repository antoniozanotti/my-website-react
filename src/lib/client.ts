import { ApolloClient, HttpLink, InMemoryCache } from "@apollo/client";
import { registerApolloClient } from "@apollo/experimental-nextjs-app-support/rsc";

const TOKEN = process.env.CONTENTFUL_ACCESS_TOKEN;
const SPACE = process.env.CONTENTFUL_SPACE_ID;
const ENVIRONMENT = process.env.CONTENTFUL_ENVIRONMENT_ID;
const URL = `https://graphql.contentful.com/content/v1/spaces/${SPACE}/environments/${ENVIRONMENT}`;

export const { getClient } = registerApolloClient(() => {
  return new ApolloClient({
    cache: new InMemoryCache(),
    link: new HttpLink({
      uri: URL,
      headers: {
        Authorization: `Bearer ${TOKEN}`,
      },
    }),
  });
});
