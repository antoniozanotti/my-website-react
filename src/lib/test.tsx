import React, { ReactElement } from "react";
import {
  render,
  RenderOptions,
  RenderResult,
  Screen,
} from "@testing-library/react";
import { Theme } from "@/domain/theme/theme/theme";
import { Providers } from "../app/providers";
import messages from "@/messages/en.json";

function AllTheProviders({ children }: { children: React.ReactNode }) {
  return (
    <Providers locale="en" messages={messages}>
      <Theme>
        <div data-testid="container">{children}</div>
      </Theme>
    </Providers>
  );
}

function customRender(
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
): RenderResult {
  return render(ui, { wrapper: AllTheProviders, ...options });
}

function firstComponent(screen: Screen) {
  return screen.getByTestId("container").firstChild;
}

export * from "@testing-library/react";
export * from "@testing-library/user-event";
export { customRender as render, firstComponent };
import "@testing-library/jest-dom";

jest.mock("next/navigation", () => {
  return {
    useRouter() {
      return { push: () => {} };
    },
    useParams: () => ({}),
  };
});
