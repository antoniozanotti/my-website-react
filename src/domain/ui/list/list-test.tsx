import { test, expect } from "@playwright/experimental-ct-react";
import { List } from "./list";

test.describe("List", () => {
  test("should render ul", async ({ mount }) => {
    const component = await mount(<List />);
    await expect(component.locator(":scope:is(ul)")).toHaveCount(1);
  });

  test("should accept className", async ({ mount }) => {
    const component = await mount(<List className='test' />);
    await expect(component).toHaveClass(/test$/);
  });

  test("should render children", async ({ mount }) => {
    const component = await mount(<List><li>Test</li></List>);
    await expect(component.locator("li")).toBeVisible();
  });
});
