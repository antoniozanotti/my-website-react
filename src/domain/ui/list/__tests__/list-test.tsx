import { screen, render, firstComponent } from "@/lib/test";
import { List } from "../list";

describe("List", () => {
  it("should render ul", async () => {
    const { container } = render(<List />);
    const component = screen.getByRole("list");

    expect(component).toBeVisible();
    expect(container).toMatchSnapshot();
  });

  it("should accept className", async () => {
    const { container } = render(<List className='test' />);
    const component = firstComponent(screen);

    expect(component).toHaveClass(/test$/);
    expect(container).toMatchSnapshot();
  });

  it("should render children", async () => {
    const { container } = render(<List><li>Test</li></List>);
    const component = screen.getByRole("listitem");

    expect(component).toBeVisible();
    expect(container).toMatchSnapshot();
  });
});
