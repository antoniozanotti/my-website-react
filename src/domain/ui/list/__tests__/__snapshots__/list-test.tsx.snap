// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List should accept className 1`] = `
<div>
  <div
    class="dark"
  >
    <div
      class="className bg-dark-2 dark:bg-dark-dark-2 text-primary dark:text-dark-primary text-c3 lg:text-c4"
    >
      <div
        data-testid="container"
      >
        <ul
          class="list-disc list-inside test"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`List should render children 1`] = `
<div>
  <div
    class="dark"
  >
    <div
      class="className bg-dark-2 dark:bg-dark-dark-2 text-primary dark:text-dark-primary text-c3 lg:text-c4"
    >
      <div
        data-testid="container"
      >
        <ul
          class="list-disc list-inside"
        >
          <li>
            Test
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;

exports[`List should render ul 1`] = `
<div>
  <div
    class="dark"
  >
    <div
      class="className bg-dark-2 dark:bg-dark-dark-2 text-primary dark:text-dark-primary text-c3 lg:text-c4"
    >
      <div
        data-testid="container"
      >
        <ul
          class="list-disc list-inside"
        />
      </div>
    </div>
  </div>
</div>
`;
